### PR TITLE
Add an example monitor: DNT Status

### PR DESCRIPTION
**Pull request by mreid-moz**
Checks the status of the Do Not Track HTTP Header.

### CODE REVIEW

File: examples/monitor_dnt.lua [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+local rows = read_config("rows") or 2880
+local sec_per_row = read_config("sec_per_row") or 60
+local ON  = 1
+local OFF = 2
+local UNK = 3
+
+local c = circular_buffer.new(rows, 3, sec_per_row, true)
+c:set_header(ON, "DNT On")
+c:set_header(OFF, "DNT Off")
+c:set_header(UNK, "DNT Unknown")
```
Comment (trink) [inline examples/monitor_dnt.lua]:
``` lua
-- remove constants

c:set_header(1, "DNT Off")
c:set_header(2, "DNT On")
c:set_header(3, "DNT Unknown")
```

...

``` lua
local item = tonumber(read_message("Fields[DNT]")) or 2

-- remove if
-- range check or pcall
c:add(ts, item+1, 1)
```

Comment (mreid-moz) [inline examples/monitor_dnt.lua]: Some of the motivation for this filter is to serve as an example - I think the general pattern of variables for cbuf column numbers + explicitly setting their names makes a good model.

Comment (trink) [inline examples/monitor_dnt.lua]: Maybe too compact for an example... 
You can just consolidate lines 24-26 with 29-31
i.e., 'local ON = c:set_header(1, "DNT On")

and you can get rid of the 'or "UNKNOWN"'

File: examples/monitor_dnt.lua [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    preserve_data = true
+--]]
+
+require "circular_buffer"
+
+local rows = read_config("rows") or 2880
+local sec_per_row = read_config("sec_per_row") or 60
+
+-- Create a circular buffer with three columns
+local c = circular_buffer.new(rows, 3, sec_per_row, true)
```
Comment (trink) [inline examples/monitor_dnt.lua]: This would have to be global to be preserved

File: examples/monitor_dnt.lua [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+.. code-block:: ini
+
+    [DNT Usage]
+    type = "SandboxFilter"
+    filename = "examples/monitor_dnt.lua"
+    message_matcher = "Type == 'telemetry'"
+    ticker_interval = 10
+    preserve_data = true
+--]]
+
```
Comment (trink) [inline examples/monitor_dnt.lua:29]:
``` lua
_PRESERVATION_VERSION = read_config("preservation_version") or 0
```

File: examples/monitor_dnt.lua [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        c:add(ts, OFF, 1)
+    else
+        c:add(ts, UNK, 1)
+    end
+
+    return 0
+end
+
+function timer_event(ns)
+    inject_payload("cbuf", "DNT Status", c)
```
Comment (trink) [inline examples/monitor_dnt.lua]: What about injecting the cbuf delta?

File: examples/monitor_dnt.lua [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    [DNTUsage]
+    type = "SandboxFilter"
+    filename = "examples/monitor_dnt.lua"
+    message_matcher = "Type == 'telemetry'"
+    ticker_interval = 10
+    preserve_data = true
+        [DNTUsage.config]
+        preservation_version = 1
+        rows = 1440
+        sec_per_row = 300
```
Comment (trink) [inline examples/monitor_dnt.lua:25]: Since this is an example we probably want docs explaining what the options do.

**Pull request merged**